### DISCUSSION TITLE
NewPipe supports Ryukyuan languages (Uchinaguchi)

### DISCUSSION THREAD

**Discussion by kuragehimekurara1** [TeamNewPipe/NewPipe#10205] [Questions] [answered]
I want to translate NewPipe into Uchinaguchi (Ryukyuan or Okinawan), but Weblate does not have Uchinaguchi, Okinawan, or Ryukyuan, so I cannot select or translate. What should I do in this case?

**Comment by TobiGr** [comment, marked as answer]
Thank you for helping with the translations!
There are two things that can / should be done.
1. Please open an issue in the [Weblate repo](https://github.com/weblateOrg/weblate) to request support for those languages
2. Add the translation manually  
    1. Create a directory named `values-ryu` (or with a different language code) in [`app/src/main/res`](https://github.com/TeamNewPipe/NewPipe/tree/dev/app/src/main/res) directory.
    2. Copy [`values/strings.xml`](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/strings.xml) into it and translate.
    3. Add your language to the in-app language chooser (e.g., see #10042).
    4. Make sure to remove all untranslated strings for your string.xml before opening a PR.
